src/dotnet/App.Wasm/wwwroot/blazorContextMenu.js

```diff
@@ -152,28 +152,62 @@ var blazorContextMenu = function (blazorContextMenu) {
         }
         addToOpenMenus(menu, menuId, e.target, target);
         const triggerDotnetRef = JSON.parse(e.currentTarget.dataset["dotnetref"]);
-        showMenuCommon(menu, menuId, x, y, e.target, triggerDotnetRef);
+        showMenuCommon(menu, menuId, x, y, e, triggerDotnetRef);
         e.preventDefault();
         if (stopPropagation) {
             e.stopPropagation();
         }
         return false;
     };
 
-    let showMenuCommon = function (menu, menuId, x, y, target, triggerDotnetRef) {
-        let isLeftHalf = x < window.innerWidth / 2;
+    let showMenuCommon = function (menu, menuId, x, y, event, triggerDotnetRef) {
+        const target = event.target;
+        const offset = 5;
+        const isLeftHalf = x < window.innerWidth / 2;
+        const isTopHalf = y < window.innerHeight / 2;
+        if (event.button === 2) {
```

Comment:
Why do you need special processing for right click?
If you need special processing for ContextMenuToggleTrigger, then i think better to render it with additional special attribute (e.g. data-context-menu-toggle=true) and test it's value in js.